xy-heatmap-quantile/xyHeatmapQuantile.js

```diff
@@ -0,0 +1,124 @@
+import * as d3 from 'd3';
+import gChartcolour from 'g-chartcolour';
+
+export function draw() {
+    let yScale = d3.scaleBand();
+    let xScale = d3.scaleBand();
+    let seriesNames = [];
+    let showValues = false;
+    let yAxisAlign = 'right';
+    let rem = 16;
+    const colourScale = d3.scaleOrdinal()
+        .domain(seriesNames);
+
+    function chart(parent) {
+        xScale.paddingInner(0);
+        yScale.paddingInner(0);
+
+        parent.attr('transform', d => `translate(0, ${yScale(d.name)})`)
+            .attr('width', xScale.bandwidth());
+
+        const block = parent.selectAll('g')
+            .data(d => d.groups)
+            .enter()
+            .append('g');
+
+        block
+            .append('rect')
+            .attr('class', (d) => {
+                if (typeof d.value === 'number') {
+                    return 'grid';
+                }
+                return 'grid noData';
+            })
+            .attr('x', d => xScale(d.name))
+            .attr('width', () => xScale.bandwidth())
+            .attr('y', d => yScale(d.name))
+            .attr('height', () => yScale.bandwidth())
+            .attr('fill', d => colourScale.range()[d.scaleCat]);
+
+        if (showValues) {
+            block
+                .append('text')
+                .attr('class', 'blockValue')
+                .attr('x', d => xScale(d.name))
+                .attr('y', d => yScale(d.name))
+                .attr('dx', (xScale.bandwidth() / 2))
```

Comment:
The parentheses here are superfluous.

Reply:
I had them cos I was dividing the bandwidth by two. Wanted to stick to bidmas rules.